### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by DaveTBlake**
Clean up a few things from the Coverity report in the music library code

### CODE REVIEW

**Comment by notspiff** [comment]
looks fine.

File: xbmc/music/dialogs/GUIDialogMusicInfo.cpp [hunk 1]
```diff
@@ -319,6 +319,7 @@ CGUIDialogMusicInfo::CGUIDialogMusicInfo(void)
   m_hasUpdatedUserrating = false;
   m_hasRefreshed = false;
   m_bArtistInfo = false;
+  m_scraperAddInfo = false;
```
Comment (Rechi) [inline xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: use default member initializer

Comment (DaveTBlake) [reply xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: But will that set all these members as required e.g. make new CFileItemList and clear string?

Comment (Rechi) [reply xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: I'm only asking to do it for the one variable you added now and yes this constructor is still called.

Comment (DaveTBlake) [reply xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: I don't see how doing this one variable (that was previously overlooked by accident) differently to the rest makes better/clearer code.

File: xbmc/music/dialogs/GUIDialogMusicInfo.cpp [hunk 2]
```diff
@@ -316,10 +316,6 @@ CGUIDialogMusicInfo::CGUIDialogMusicInfo(void)
   m_albumSongs = new CFileItemList;
   m_loadType = KEEP_IN_MEMORY;
   m_startUserrating = -1;
```
Comment (notspiff) [inline xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: remove.

File: xbmc/music/dialogs/GUIDialogMusicInfo.cpp (hunk 1 above)
Comment (DaveTBlake) [reply xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: Ah OK, this got missed from global change to defaut member initializers. I will do all bools

File: xbmc/music/dialogs/GUIDialogMusicInfo.cpp (hunk 2 above)
Comment (DaveTBlake) [reply xbmc/music/dialogs/GUIDialogMusicInfo.cpp]: Doh, commited without save!

**Comment by DaveTBlake** [comment]
jenkins build this please

**Pull request merged**